backend/layers/persistence/persistence.py

```diff
@@ -114,23 +114,26 @@ def _row_to_collection_version(self, row: Any, canonical_collection: CanonicalCo
             created_at=row.created_at,
             schema_version=row.schema_version,
             canonical_collection=canonical_collection,
+            datasets_custom_ordered=row.datasets_custom_ordered,
         )
 
     def _row_to_collection_version_with_datasets(
         self, row: Any, canonical_collection: CanonicalCollection, datasets: List[DatasetVersion]
     ) -> CollectionVersionWithDatasets:
+        sorted_datasets = self._sort_datasets(row, datasets)
```

Comment:
Adding sort here in the persistence layer before dataset version IDs (i.e. dataset order) are replaced with dataset versions.

Reply:
If we go by the original design, I would argue that this belongs to the business logic.

Reply:
That said, I'm not suggesting that we change it now, since it will be quite a few changes - just a reminder that we diverged quite significantly from the original design and the difference between the two layers is now less marked.